### DISCUSSION TITLE
Question regarding TodoMVC React example

### DISCUSSION THREAD

**Discussion by phuoccss** [statelyai/xstate#1298] [General] [answered]
Hi. In the [TodoMVC React example](https://codesandbox.io/s/xstate-todomvc-33wr94qv1), I notice that todo item data are in both todosMachine and individual todoMachine.

Is there a reason for this as it seems to cause unnecessary re-render? E.g. when I edit a todo, I don't expect both todo list and other todos to rerender.

I'm still new to XState so please ignore if I missed anything.

**Comment by davidkpiano** [comment, marked as answer]
This is using an older version of XState and `@xstate/react`. Keep on the lookout for an updated (and more optimized) version when `@xstate/react` version 1 comes out!
